### PR TITLE
nlohmann_json: add 3.9.0 (and implicit_conversions option)

### PR DESCRIPTION
**Pull request by theodelrieu**
Specify library name and version:  **nlohmann_json/3.9.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/nlohmann_json/all/conanfile.py [hunk 1]
```diff
@@ -13,16 +13,26 @@ class NlohmannJsonConan(ConanFile):
...
     }
     default_options = {
-        "multiple_headers": False
+        "multiple_headers": False,
+        "implicit_conversions": True
     }
 
+    @property
+    def _can_disable_implicit_conversions(self):
+        return tools.Version(self.version) >= "3.9.0"
```
Comment (danimtb) [inline recipes/nlohmann_json/all/conanfile.py:26]: so this option is not available in versions lower than this one?

Comment (theodelrieu) [reply recipes/nlohmann_json/all/conanfile.py:26]: Yes, it has been added by https://github.com/nlohmann/json/pull/1559

Comment (danimtb) [reply recipes/nlohmann_json/all/conanfile.py:26]: then this option should not be part of the package id for lower versions

Comment (theodelrieu) [reply recipes/nlohmann_json/all/conanfile.py:26]: I do remove it in `config_options`, isn't it enough?

Comment (danimtb) [reply recipes/nlohmann_json/all/conanfile.py:26]: gotcha

**Pull request merged**